### PR TITLE
Allow plugins to extend or replace the LoginForm

### PR DESCRIPTION
**Pull request by smashwilson**
I've added two `{% pluginblock %}` sections to the login view to allow plugins to either completely replace, or add login buttons to, the login form.

screen shot 2016-03-28 at 9 57 20 am

screen shot 2016-03-28 at 9 54 43 am

I'm currently hacking around this by [abusing `LoggedOutFillContent`](https://github.com/smashwilson/strider-github-oauth/blob/master/webapp.js#L30-L33), but that only works on `/`, not `/login`.

### CODE REVIEW

File: lib/views/partials/loginform.html [hunk 1]
```diff
@@ -4,28 +4,32 @@
...
+              </div>
+            </form>
+            <p><strong><a href="/register">I have an invite code.</a></strong></p>
+            <p><a id="forgot-password-link" href="/forgot">Forgot your password?</a></p>
+          {% endpluginblock %}
         </div>
       </div>
-   </div>
+    </div>
+  </div>
```
Comment (smashwilson) [inline lib/views/partials/loginform.html]: Apologies for the whitespace diffs here - I tidied it up a bit to use 2-space indents; it had been mixed 2-space and 4-space.

**Comment by knownasilya** [comment]
What's the usecase for completely replacing the login form?

**Comment by smashwilson** [comment]
Oh, I've been doing something similar to avoid confusion from users with our [heavily opinionated GitHub OAuth plugin](https://github.com/smashwilson/strider-github-oauth). We back Strider users entirely with membership in GitHub organizations and teams - Strider users are created on first login, admin rights are granted to organization admins or membership on an admin team. None of our user accounts [even have real passwords](https://github.com/smashwilson/strider-github-oauth/blob/master/authorization.js#L82), so I hid the form entirely to have one clear path to authenticate.

I could easily be persuaded to take it out, though.

**Comment by knownasilya** [comment]
I would love to add github-oauth to strider proper, so that would be the path I'd desire above adding a patch point that is more of a work-around.

**Comment by knownasilya** [comment]
I added your plugin to https://github.com/Strider-CD/strider/wiki/List-of-Plugins#strider-github-oauth

**Comment by smashwilson** [comment]
> I would love to add github-oauth to strider proper, so that would be the path I'd desire above adding a patch point that is more of a work-around.

Reasonable; I've cut the `LoginForm` pluginblock. `ExtraLoginButton` could still be useful for any number of alternate login flows, though.

I noticed that most of the hooks were already there to inject arbitrary passport handlers and so on. Is there an issue to track what's missing? I could swear I found one before but search is failing me now.

> I added your plugin to https://github.com/Strider-CD/strider/wiki/List-of-Plugins#strider-github-oauth

:+1:

It's likely a bit too specialized still (and it needs things like "better error messages and not just a stack trace if you don't have access") but it's a start.

**Comment by knownasilya** [comment]
Haven't seen an issue along those lines, so your guess is as good as mine.

**Pull request merged**